eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -144,18 +142,14 @@ public ClusteringWriter(MongoTemplate mongoTemplate,
     private void getSVOEWithMergeAndRSSplitCandidates() {
         List<String> MERGE_AND_SPLIT_EVENTS = Arrays.asList(EventType.RS_MERGE_CANDIDATES.name(),
                                                             EventType.RS_SPLIT_CANDIDATES.name());
-        AggregationOptions aggregationOptions = Aggregation.newAggregationOptions().cursorBatchSize(1000)
-                .allowDiskUse(true).build();
+        Query queryOperations = query(where("eventType").in(MERGE_AND_SPLIT_EVENTS)
+                                                        .and("inactiveObjects").elemMatch(where("seq").is(assembly)));
         mergeCandidateSVOE = new HashMap<>();
         rsSplitCandidateSVOE = new HashMap<>();
 
         try (
-                CloseableIterator<SubmittedVariantOperationEntity> svoes = mongoTemplate.aggregateStream(
-                        Aggregation.newAggregation(
-                                Aggregation.match(where("eventType").in(MERGE_AND_SPLIT_EVENTS)
-                                        .and("inactiveObjects").elemMatch(where("seq").is(assembly))))
-                        .withOptions(aggregationOptions), SubmittedVariantOperationEntity.class.getSimpleName(),
-                SubmittedVariantOperationEntity.class)) {
+                CloseableIterator<SubmittedVariantOperationEntity> svoes = mongoTemplate.stream(
+                        queryOperations, SubmittedVariantOperationEntity.class)) {
```

Comment:
This was changed from `find` to `aggregate` [here](https://github.com/EBIvariation/eva-accession/pull/388), but I can't see why it's necessary - if I understand the context correctly, that PR was more about using an iterator to alleviate memory issues, which should be achievable without using aggregations.

Reply:
Actually it looks like this aggregate simply did not work, and this is the origin of the duplicate keys - will write it up in the ticket.